pycyphal/transport/udp/_ip/_v4.py

```diff
@@ -27,18 +31,16 @@ class IPv4SocketFactory(SocketFactory):
     a node-ID that maps to the broadcast address for the subnet is unavailable.
     """
 
-    def __init__(self, local_ip_address: IPv4Address):
-        if not isinstance(local_ip_address, IPv4Address):  # pragma: no cover
-            raise TypeError(f"Unexpected IP address type: {type(local_ip_address)}")
-        self._local = local_ip_address
+    def __init__(self, local_ip_addr: typing.Union[ipaddress.IPv4Address, ipaddress.IPv6Address]):
```

Comment:
An IPv4 socket factory cannot be used if the local address is IPv6.

```suggestion
    def __init__(self, local_ip_addr: ipaddress.IPv4Address):
```